Fix: Display term names instead of term IDs for product_cat and product_shipping_class on product page

arifulhoque7 opened the pull request:
* **New Features**
  * Enhanced support for WooCommerce taxonomies, including improved handling of product categories, tags, shipping classes, and attributes.
  * Automatic creation and assignment of missing WooCommerce terms during product updates.

* **Bug Fixes**
  * Improved data consistency and reliability through stricter type comparisons and better data normalization.

* **Refactor**
  * Streamlined taxonomy and attribute handling for WooCommerce products, resulting in more predictable and safer behavior.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Enhanced product attribute display with custom form-based labels for WooCommerce products
  * Improved support for product taxonomies including categories, tags, and shipping classes

* **Bug Fixes**
  * Strengthened validation checks for reCAPTCHA verification
  * Better product attribute metadata handling and WooCommerce product data synchronization
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

arifulhoque7 committed:
Fix: Display term names instead of term IDs for product_cat and product_shipping_class on product page

sapayth commented on includes/Traits/FieldableTrait.php:
```diff
@@ -439,101 +453,207 @@ public function update_post_meta( $meta_vars, $post_id ) {
-                        $tax = [ $tax ];
+            // Handle other WooCommerce taxonomies if WooCommerce is enabled
+            if ( $is_woo_taxonomy && (
+                'pa_' === substr( $taxonomy_name, 0, 3 ) || 
```
do not check this way. use ` This taxonomy is a WooCommerce attribute` option from form builder field setting

arifulhoque7 commented:
Already done based on business requirement , can close for now. Related [issue](https://github.com/weDevsOfficial/wpuf-pro/pull/907)

arifulhoque7 committed:
fix: PR review incorporate

arifulhoque7 committed:
fix: code review issues:

arifulhoque7 committed:
Merge branch 'weDevsOfficial:develop' into fix/product-meta-display-term-labels

arifulhoque7 committed:
fix: issues for check the product_terms

arifulhoque7 committed:
fix: data not saving

arifulhoque7 commented:
> Detected issues in the following video! @arifulhoque7 vai
> 
>  CleanShot.2025-08-01.at.16.04.38.1.mp4

please check not it is okay

arifulhoque7 committed:
fix: redundent code in form builder css

arifulhoque7 committed:
fix: code optimization

arifulhoque7 committed:
fix: error log removed

arifulhoque7 committed:
Merge remote-tracking branch 'upstream/develop' into fix/product-meta-display-term-labels

# Conflicts:
#	admin/form-builder/assets/js/form-builder.js
#	assets/css/admin/form-builder.css
#	assets/js/wpuf-form-builder.js
#	includes/Traits/FieldableTrait.php
#	wpuf-functions.php

arifulhoque7 committed:
style: strip trailing whitespace in woocommerce-attribute-labels.php (CodeRabbit)

arifulhoque7 commented on assets/css/admin/form-builder.css:
```diff
@@ -843,6 +843,63 @@ html {
+  --tw-text-opacity: 1;
+  color: var(--fallback-bc,oklch(var(--bc)/var(--tw-text-opacity)));
+}
+
```
Skipping — this is generated Tailwind output (the duplicate `transition-property` lines are intentional `-webkit-backdrop-filter`/`backdrop-filter` progressive-enhancement fallbacks). The file is pulled from `develop` and regenerated at build, so a hand edit here would be overwritten.

arifulhoque7 commented on assets/css/admin/form-builder.css:
```diff
@@ -1305,6 +1752,60 @@
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
+  transition-duration: 200ms;
+  box-shadow: rgba(0, 0, 0, 0.25) 0px 25px 50px -12px;
```
Skipping — same as the .wpuf-badge case: generated Tailwind vendor-prefix fallbacks, regenerated at build. Not hand-editable source.

arifulhoque7 commented on assets/css/admin/form-builder.css:
```diff
@@ -843,6 +843,63 @@ html {
+  transition-property: color, background-color, border-color, text-decoration-color, fill, stroke, opacity, box-shadow, transform, filter, backdrop-filter, -webkit-backdrop-filter;
+  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
+  transition-timing-function: cubic-bezier(0, 0, 0.2, 1);
+  transition-duration: 200ms;
```
Skipping — generated Tailwind output; the duplicate `transition-timing-function` is part of the compiled fallback chain and is regenerated at build.

arifulhoque7 commented on includes/functions/woocommerce-attribute-labels.php:
```diff
@@ -0,0 +1,53 @@
+    }
+    
+    return $label;
+}
```
Done — stripped trailing whitespace and fixed the EOF newline in woocommerce-attribute-labels.php.

arifulhoque7 commented on includes/Traits/FieldableTrait.php:
```diff
@@ -438,106 +452,761 @@ public function update_post_meta( $meta_vars, $post_id ) {
-                                // woocommerce check
-                                if ( isset( $taxonomy['woo_attr'] ) && $taxonomy['woo_attr'] == 'yes' && ! empty( $_POST[ $taxonomy['name'] ] ) ) {
-                                    $woo_attr[ $taxonomy['name'] ] = $this->woo_attribute( $taxonomy );
-                                }
-                            }
-                        } // hierarchical
-                    } // is text
-                } // is object tax
-            } // isset tax
+        if ( ! empty( $term_names ) ) {
+            wp_set_post_terms( $post_id, $term_names, $taxonomy_name, false );
+        }
+    }
 
-            else {
-                if ( isset( $taxonomy_name ) && 0 === absint( $taxonomy_name ) ) {
-                    wp_set_post_terms( $post_id, $taxonomy_name, $taxonomy['name'] );
+    /**
+     * Handle shipping class
+     */
+    protected function handle_shipping_class( $post_id, $taxonomy_name, $posted_terms ) {
+        // Clear existing shipping classes
+        wp_set_object_terms( $post_id, [], 'product_shipping_class' );
+        
+        if ( empty( $posted_terms ) || '0' === $posted_terms || '-1' === $posted_terms ) {
+            // No shipping class selected
+            return;
+        }
+        
+        // Handle both IDs and slugs
+        if ( is_numeric( $posted_terms ) ) {
+            $term = get_term( $posted_terms, 'product_shipping_class' );
+            if ( $term && ! is_wp_error( $term ) ) {
+                wp_set_object_terms( $post_id, $term->slug, 'product_shipping_class', false );
+            }
+        } else {
+            // It's already a slug or name
+            wp_set_object_terms( $post_id, $posted_terms, 'product_shipping_class', false );
+        }
+    }
```
Keeping `$taxonomy_name` — these handlers are invoked through a uniform dispatch (`$this->$method( $post_id, $taxonomy_name, $posted_terms )`), so every handler must share the same signature even where the body doesn't reference it. Removing the param would break the dispatcher.

arifulhoque7 committed:
fix: sanitize posted taxonomy terms and use strict in_array (review)

- Sanitize $_POST[ $taxonomy_name ] (array_map sanitize_text_field / sanitize_text_field) after wp_unslash before passing to term functions
- Add strict true to all in_array() calls in the trait (PHPCS zero-tolerance)
